md_server/session.py

```diff
@@ -1,26 +1,57 @@
 import asyncio
 import logging
+from mdlib import md_pb2
+from mdlib.socket_utils import LengthReader, LengthWriter
+from mdlib.db_utils import get_db_md5, MDActions
 
 
 class Session(object):
-    def __init__(self, server, reader: asyncio.StreamReader, writer: asyncio.StreamWriter, client_id, db_name):
+    def __init__(self, server, reader: asyncio.StreamReader, writer: asyncio.StreamWriter):
         self.server = server
-        self.reader = reader
-        self.writer = writer
-        self.client_id = client_id
-        self.db_name = db_name
+        self.reader = LengthReader(reader)
+        self.writer = LengthWriter(writer)
+        self.client_id = None
+        self.db_name = None
+        self.db_actions = None # can exist only after db_name is known
 
-        self.push_db()
         self.session_task = asyncio.create_task(self.handle_session())
 
     async def handle_session(self):
+        await self.get_db_info()
+        await self.push_db()
         while True:
-            request = await self.reader.read(2)  # Somehow should know how much to read
+            self.db_actions = MDActions(self.server.directory, self.db_name)
+            request = await self.reader.read()
             logging.debug(f"Got request from server {request}")
             self.server.handle_session_request(self.db_name, request)
 
-    def push_db(self):
-        pass
+    async def get_db_info(self):
+        db_info = md_pb2.DBInfo()
+        db_info.ParseFromString(await self.reader.read())
+        self.client_id = db_info.client_id
+        self.db_name = db_info.db_name
+        logging.info(f"Client {self.client_id} connected to db {self.db_name}")
+
+    async def push_db(self):
```

Comment:
A little bit messy but for now it is good

Reply:
this need to be changed to be a protobuf action in ACTIONS
TODO..